storage-node/src/services/webApi/controllers/stateApi.ts

```diff
@@ -158,7 +159,7 @@ export async function getStatus(req: express.Request, res: express.Response<Stat
     downloadBuckets,
     sync,
     cleanup,
-    queryNodeStatus: await getQueryNodeStatus(qnApi),
+    queryNodeStatus: await getQueryNodeStatus(api, qnApi),
```

Comment:
how does express handle the response if we get an uncaught exception here? Is it handled by https://github.com/Joystream/joystream/blob/691e35f6b846fb07637fefe9372e4a49ff0543b5/storage-node/src/services/webApi/app.ts#L109 ?

Reply:
I guess this is a similar point made in https://github.com/Joystream/joystream/pull/5001#pullrequestreview-1798426461

Reply:
Actually, error handler middleware is not really used/called in the `storage-node` (even in all the previous versions), so whenever the unhandled exception occurred in the route handlers, it caused the app to crash. 

If async route handlers throw then its the [responsibility](https://expressjs.com/en/guide/error-handling.html#:~:text=For%20errors%20returned%20from%20asynchronous%20functions%20invoked%20by%20route%20handlers%20and%20middleware%2C%20you%20must%20pass%20them%20to%20the%20next()%20function%2C%20where%20Express%20will%20catch%20and%20process%20them.%20For%20example%3A) of the app to pass this error to the `next()` function which then will be handled by the default error handler

I have [fixed](https://github.com/Joystream/joystream/pull/5001/commits/7d5d37fd25a955ea06adcda5779acd22361e2c43) this for storage-node, for `distributor-node` this is not a problem as error thrown from the route handlers are already passed to the [next()](https://github.com/zeeshanakram3/joystream/blob/master/distributor-node/src/services/httpApi/HttpApiBase.ts#L30) function

Reply:
Addressed in [e2ce216](https://github.com/Joystream/joystream/pull/5001/commits/e2ce21622463a09f62c639ab6eee332d0678ecfd)